macros/utils/table_operations/create_or_replace.sql

```diff
@@ -4,17 +4,24 @@
 
 {# Snowflake and Bigquery #}
 {% macro default__create_or_replace(temporary, relation, sql_query) %}
-    {%- do run_query(dbt.create_table_as(temporary, relation, sql_query)) %}
+    {% do run_query(dbt.create_table_as(temporary, relation, sql_query)) %}
 {% endmacro %}
 
 {% macro redshift__create_or_replace(temporary, relation, sql_query) %}
-    {%- do dbt.drop_relation_if_exists(relation) -%}
-    {%- do run_query(dbt.create_table_as(temporary, relation, sql_query)) %}
+    {% do dbt.drop_relation_if_exists(relation) %}
+    {% do run_query(dbt.create_table_as(temporary, relation, sql_query)) %}
     {% do adapter.commit() %}
 {% endmacro %}
 
+{% macro postgres__create_or_replace(temporary, relation, sql_query) %}
+    {% do run_query("BEGIN") %}
+    {% do dbt.drop_relation_if_exists(relation) %}
+    {% do run_query(dbt.create_table_as(temporary, relation, sql_query)) %}
+    {% do run_query("COMMIT") %}
```

Comment:
I'm wondering if the Postgres implementation specifically should be in a transaction if the rest are not, WDYT?

Reply:
I don't have an alternative really.
Calling `adapter.commit()` fails because it says that I haven't started a transaction.
And not doing so just doesn't commit.

The docs says that it's okay to do so however.
<img width="873" alt="image" src="https://user-images.githubusercontent.com/30181361/212559123-e5d074cf-bda9-4faa-87be-43e5fbffd6a0.png">

Reply:
OK got it
Didn't realize it was needed by the implementation